code/src/reducers/maze.js

```diff
@@ -0,0 +1,105 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const maze = createSlice({
+  name: 'maze',
+  initialState: {
+    username: '',
+    coordinates: '',
+    description: '',
+    moves: null,
+    isLoading: false,
+    error: null,
+    gameState: '',
+  },
+  reducers: {
+    setUsername: (store, action) => {
+      store.username = action.payload
+    },
+    setMoves: (store, action) => {
+      store.moves = action.payload
+    },
+    setDescription: (store, action) => {
+      store.description = action.payload
+    },
+    setLoading: (store, action) => {
+      store.isLoading = action.payload
+    },
+    setCoordinates: (store, action) => {
+      store.coordinates = action.payload
+    },
+    setError: (store, action) => {
+      store.error = action.payload
+    },
+    setGameState: (store) => {
+      if (store.coordinates === '0,0') {
+        const setGameState = 'lose'
+        store.gameState = setGameState
+      } else if (store.coordinates === '1,3') {
+        const setGameState = 'win'
+        store.gameState = setGameState
+      }
+    },
+    restartGame: (store) => {
+      const setInitialState = ''
+      store.username = setInitialState
+      store.gameState = setInitialState
+    }
+  }
+})
+
+export default maze
+
+export const firstFetch = () => {
```

Comment:
Namingthe thunks first and second fetch is not very descriptive of what they do. ;-)